GitUI/GitUIExtensions.cs

```diff
@@ -113,11 +113,11 @@ void OpenWithDiffTool()
                         : diffOfConflict;
                 }
 
-                if (file.IsSubmodule
-                    && file.GetSubmoduleStatusAsync() is Task<GitSubmoduleStatus> task)
+                if (file.IsSubmodule)
                 {
                     // Patch already evaluated
-                    var status = ThreadHelper.JoinableTaskFactory.Run(() => task);
+                    var status = ThreadHelper.JoinableTaskFactory.Run(
+                        () => file.GetSubmoduleStatusAsync() ?? Task.FromResult<GitSubmoduleStatus?>(null));
```

Comment:
Perhaps:
```suggestion
                    GitSubmoduleStatus? status = file.GetSubmoduleStatusAsync()?.GetAwaiter().GetResult();
```

BTW: `GetSubmoduleStatusAsync` turned into a property  would avoid the misleading call `()`.

Reply:
I used what @drewnoakes suggested
This is hopefully a short term fix only
Will test this some time...

Reply:
That change hangs GE so I will not change